### PR TITLE
restructuring the code

### LINKED ISSUE #35: rescructure the code

### PR DESCRIPTION
**Pull request by b-biswas**
Fixes #35 

In the current outline, I am separating the code into the following steps: 

- [x] object detection
- [x] object extraction 
- [x] load the trained model(s) within the 'model.py' file
- [x] deblender (rename debvader.py to deblender.py [TODO])
- [x] position optimization

### CODE REVIEW

File: debvader/io_utils.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    data_path = pkg_resources.resource_filename("debvader", "data/")
+    loading_path = os.path.join(data_path, "weights/", survey, "not_normalised/loss/")
+    print(loading_path)
+    latest = tf.train.latest_checkpoint(loading_path)
+    net.load_weights(latest)
+
+    if return_encoder_decoder_z:
+        return net, encoder, decoder, z
+    else:
+        return net
```
Comment (aboucaud) [inline debvader/io_utils.py:58]:
```suggestion
    return net
```

File: debvader/io_utils.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        conv_activation=None,
+        dense_activation=None,
+    )
+
+    # Define the loss function
+    def vae_loss(x, x_decoded_mean):
+        # xent_loss = K.mean(
+        #    K.sum(K.binary_crossentropy(x, x_decoded_mean), axis=[1, 2, 3])
+        # )
+        return -x_decoded_mean.log_prob(x)  # xent_loss
```
Comment (aboucaud) [inline debvader/io_utils.py:36]: That could go in `debvader.metrics` along with the `mse` you recently defined.

File: debvader/position_optimization.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    parameters:
+        field image: image of the entire field of galaxy to be deblended.
+        output_images_mean_padded: predicted image of the galaxy that is to be optimized.
+        galaxy_distances_to_center: distance of the predicted galaxy from the center, as detected by the detection algorithm.
+    """
+
+    assert method in ["scipy-minimize"]
+
+    if method == "scipy-minimize":
+
```
Comment (aboucaud) [inline debvader/position_optimization.py:24]: I don't see the point in "preparing" for other methods.
Right now you only have a single method so don't go through the burden of adding checks and if statements. I would suggest you remove that part and add it again along with support for new optimization schemes.

**Review by aboucaud — COMMENTED** [review]
This refactoring looks good. Really how I intended it to be done.

I'll add one more request before approving this PR : a common thing to adopt in python packaging is to give short names for the submodules, in order to navigate easily.
Therefore I would suggest the renaming of `debvader`submodules as: `extraction`, `detection`, `optimization`, and move the content of the current `io_utils` to `model`, where it belongs naturally.

**Comment by b-biswas** [comment]
@aboucaud I resolved your comments and renamed the debvader.py file as well. 
If it looks ok to you know I will go ahead and merge it

File: debvader/metrics.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import numpy as np
+
+
+def mse(img1, img2):
+    return np.mean(np.square(img1 - img2))
+
+
+# Define the loss as the log likelihood of the distribution on the image pixels
+def vae_loss(x, x_decoded_mean):
+    return -x_decoded_mean.log_prob(x)
```
Comment (aboucaud) [inline debvader/metrics.py:10]: Two things : 
- now that the loss has been separated from the code it needs to be more explicit and renaming of the second argument
- comment => docstring

+ its the **negative** log likelihood of the distribution **of** the image pixels

**Review by aboucaud — COMMENTED** [review]
A few more comments along the way :) 

Nearly there :)

**Comment by b-biswas** [comment]
> A few more comments along the way :)
> 
> Nearly there :)

I added the docstrings!
But sometime later I intend to have a pre-commit hook to properly follow a documentation style

**Review by aboucaud — APPROVED** [review]
